### ISSUE TITLE
"Proposed price" field has no label

### ISSUE THREAD

**Issue by toolness** [18F/calc#547]
The HTML looks like this:

``` html
<div class="proposed-price">
  <input type="number" step="0.01" name="proposed-price" placeholder="Proposed price" class="two-decimal-places" />
  <button class="button-primary go" />Go</button>
</div>
```

It needs a `<label>` associated with it to be accessible.

Related: #546

**Comment by toolness** [comment]
Fixed by #558!

**Comment by toolness** [comment]
Ack, wait... I just noticed that #558 sets the `for` attribute of the label to the `name` attribute of the input... but it's supposed to be the `id` of the input, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label).  Which is kind of annoying because it means we have to give the input an `id`. 😞 

Also, a weird thing I just discovered is that VoiceOver apparently uses the placeholder text to identify the form input, which means the `<label>` probably isn't necessary on VoiceOver... but it's still a best practice--I guess not all screen readers may consult the placeholder text?--so we should probably Do The Right Thing, I guess.

Reopening this issue!

**Comment by hbillings** [comment]
Reopening because derp.
